components/cdc/src/observer.rs

```diff
@@ -103,9 +103,11 @@ impl CdcObserver {
 impl Coprocessor for CdcObserver {}
 
 impl<E: KvEngine> CmdObserver<E> for CdcObserver {
+    #[allow(unused_assignments)]
     fn on_prepare_for_apply(&self, cdc: &ObserveHandle, rts: &ObserveHandle, region_id: u64) {
-        *self.last_batch_observing.borrow_mut() = cdc.is_observing();
-        if !*self.last_batch_observing.borrow() {
+        let mut last_batch_observing = *self.last_batch_observing.borrow_mut();
```

Comment:
This is wrong, `last_batch_observing` is a copy instead of a reference to `self.last_batch_observing`.

Why not
```
let is_observing = cdc.is_obsering();
*self.last_batch_observing.borrow_mut() = is_observing;
if !is_observing {
...
```
```